orchagent/portsorch.cpp

```diff
@@ -2927,6 +3055,49 @@ void PortsOrch::doPortTask(Consumer &consumer)
                         SWSS_LOG_NOTICE("Set port %s AutoNeg from %d to %d", alias.c_str(), p.m_autoneg, an);
                         p.m_autoneg = an;
                         m_portList[alias] = p;
+                        m_portStateTable.hdel(p.m_alias, "rmt_adv_speeds");
+                        updatePortStatePoll(p, PORT_STATE_POLL_AN, (an > 0));
+                    }
+                }
+
+                if (!lt_str.empty() && (p.m_type == Port::PHY))
+                {
+                    if (link_training_mode_map.find(lt_str) == link_training_mode_map.end())
```

Comment:
Similar to AN, I don't see a lt capability query here. Please add

Reply:
This is intentionally removed at the earlier commit
1. LT is similar to AN, it's a port capability available on a per-port basis, not system-wide on a single switch ASIC
2. For a port capability available on a per-port basis, SAI_PORT_ATTR_SUPPORTED_XXXX should be introduced into saiport.h
3. As per the latest comments from Rita, the SAI spec. was finalized around March time frame, it's too late for 202205 right now
4. Since this is a port capability available on a per-port basis, using SAI_PORT_ATTR_LINK_TRAINING_ENABLE in sai_query_attribute_capability  is not going to work, and during the SAI subgroup meeting a few months back, there was a conclusion that per-port capability query should be tracked in a separated PR, as all the existing attributes should all be migrated, not just the new attribute for LT.

Reply:
For your reference, the function prototype of sai_query_attribute_capability is as follows, it does not work on a per-port basis
https://github.com/opencomputeproject/SAI/blob/master/inc/saiobject.h#L221
```
sai_status_t sai_query_attribute_capability(
        _In_ sai_object_id_t switch_id,
        _In_ sai_object_type_t object_type,
        _In_ sai_attr_id_t attr_id,
        _Out_ sai_attr_capability_t *attr_capability);
```

Reply:
Can you please add a comment to make sure once SAI is there we revisit and add the check? It is important that we ensure it so that devices and specific ports where it is not supported will have check in place to avoid SAI failure thereby prevent orchagent crash

Reply:
Done